tracing/src/main/java/com/palantir/remoting2/tracing/Tracers.java

```diff
@@ -96,11 +100,14 @@ public static Runnable wrap(Runnable delegate) {
         @Override
         public V call() throws Exception {
             Trace originalTrace = Tracer.copyTrace();
+            String originalMdcTraceIdValue = MDC.get(MDC_KEY);
             Tracer.setTrace(trace);
+            MDC.put(MDC_KEY, trace.getTraceId());
```

Comment:
Instead of having to figure out all the places to inject MDC context we could:
 1. use the custom layout
 2. provide a logback extension (there's a standard extension point) to do the extraction

For instance, this doesn't fix that UserId and SessionId will carry into the logging. The problem is broader than traceIds and fixing it piecemeal doesn't seem ideal.